zebra-state/src/service/finalized_state/disk_format/shielded.rs

```diff
@@ -140,3 +149,39 @@ impl FromDisk for orchard::tree::NoteCommitmentTree {
             .expect("deserialization format should match the serialization format used by IntoDisk")
     }
 }
+
+impl FromDisk for (block::Height, block::Height, sapling::tree::Node) {
+    fn from_bytes(bytes: impl AsRef<[u8]>) -> Self {
+        bincode::DefaultOptions::new()
```

Comment:
I tried to do this in https://github.com/ZcashFoundation/zebra/pull/7334/commits/62324d6732283fae520a8b4961af5cc809a1578c, but wasn't sure how else to use `Serialize`/`Deserialize`.

Reply:
`serde` is the problem here, because the data it serializes depends on the inner structure of the types. So please don't use `serde` at all.

(It's ok to use `serde` for RPC types where we control the data structures entirely, and we have snapshot tests for the RPC output format.)